Feat/week-2

Trudy2645 opened the pull request:
# 변경점 👍
- 댓글 및 대댓글 기능 구현
- 멘토 피드백 반영
- 통합 및 유닛 테스트

 # 스크린샷 🖼
<img width="1438" height="891" alt="스크린샷 2025-11-07 오후 4 22 59" src="https://github.com/user-attachments/assets/a661c98f-75bf-46b6-bfa1-8aedb46c2bc7" />
<img width="1488" height="919" alt="스크린샷 2025-11-07 오후 4 23 17" src="https://github.com/user-attachments/assets/260bca98-2152-4dcb-800a-f724376e8b07" />
<img width="677" height="751" alt="스크린샷 2025-11-07 오후 12 46 27" src="https://github.com/user-attachments/assets/5fb165d8-07ec-4235-a1de-739ba1b0b6e7" />
<img width="600" height="832" alt="스크린샷 2025-11-07 오후 12 48 23" src="https://github.com/user-attachments/assets/62fa3b0c-b2bf-4d45-9f8d-cb5bd6a188ec" />
<img width="537" height="588" alt="스크린샷 2025-11-07 오후 12 48 47" src="https://github.com/user-attachments/assets/8899ff81-a0cf-4c30-80f7-9c7170814140" />

Trudy2645 committed:
Feat: CRUD 기능 구현

Trudy2645 committed:
Chore: 불필요한 주석 정리

Trudy2645 committed:
Chore: 멘토 피드백 반영: @Column TEXT 로 지정, Setter는 캡슐화를 위해 수정, DTO에 id 포함하여 구조 단순화

Trudy2645 committed:
Feat: 댓글 및 대댓글 기능 추가

Trudy2645 committed:
Feat: 통합 테스트 및, service 테스트 코드 추가

Trudy2645 committed:
Chore: 변수 수정

Martinel2 commented on src/main/java/com/example/devSns/dto/CommentRequestDto.java:
```diff
@@ -0,0 +1,12 @@
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class CommentRequestDto {
+    private String content;
+    private Long postId;
+    private Long parentId; // 대댓글인 경우 부모 댓글의 ID, 아니면 null
+}
```
DTO를 구성할 때도 Setter를 사용하게 되면 의도치않게 값이 변경될 가능성을 남겨두는 것이기 때문에, Record 를 사용하거나, 변수에 final을 사용한 후, 생성자 패턴을 사용하여 구성하는 것이 안전할 것 같아요!

Martinel2 commented on src/main/java/com/example/devSns/dto/CommentResponseDto.java:
```diff
@@ -0,0 +1,33 @@
+public class CommentResponseDto {
+    private Long id;
+    private String content;
+    private List<CommentResponseDto> children; // 대댓글 목록
```
여기는 아래 from 함수가 있으니, final을 사용하여 불변성만 챙겨줘도 괜찮을 것 같아요

Martinel2 commented on src/main/java/com/example/devSns/repository/CommentRepository.java:
```diff
@@ -0,0 +1,14 @@
+public interface CommentRepository extends JpaRepository<Comment, Long> {
+
+    // 특정 게시글의 모든 최상위 댓글을 조회 (대댓글이 아닌 댓글)
+    List<Comment> findByPostIdAndParentIsNull(Long postId);
```
최상위 댓글만을 조회하는 방식을 채택한 이유가 있을까요? 
(정답이 없는 문제이고 단순한 궁금증으로 질문한 것입니당..)

그리고 나중에 필요하게될 기능들엔 어떤 것들이 존재할까요?

Martinel2 commented on src/main/java/com/example/devSns/service/CommentService.java:
```diff
@@ -0,0 +1,76 @@
+    // 3. 댓글 수정
+    @Transactional
+    public Comment updateComment(Long id, String content) {
+        Comment comment = commentRepository.findById(id)
+                .orElseThrow(() -> new IllegalArgumentException("Comment not found"));
```
아래의 댓글 삭제에도 같은 로직이 들어가있는 중복 코드인 것 같아요! 해당 댓글이 존재하는지 확인하는 로직을 분리해서 작성하면 추후에 수정이 필요할 때, 더욱 쉽게 수정하실 수 있을거에요.

Martinel2 commented on src/main/java/com/example/devSns/service/CommentService.java:
```diff
@@ -0,0 +1,76 @@
+    // 1. 댓글 생성
+    @Transactional
+    public Comment createComment(CommentRequestDto requestDto) {
+        Post post = postRepository.findById(requestDto.getPostId())
+                .orElseThrow(() -> new IllegalArgumentException("Post not found"));
```
댓글부분을 먼저 찾았는데, 여기에도 아래의 댓글 조회 기능에서 같은 방식으로 해당 게시글이 존재하는지 확인하는 기능이 존재하고 있어요!

Martinel2 reviewed (approved):
2주차도 수고하셨습니다!! 코드를 보니 테스트 코드까지 처음이지만 아주 잘 구성하신 것 같아요!
도전해볼 점이 있다면, 먼저 테스트 코드를 작성하고, 그 테스트가 성공하도록 코드를 짜는 TDD방식을 적용해봐도 괜찮을 것 같아요. 
그리고 테스트 코드를 작성하실 때, 조금 더 작은 범위의 테스트들도 존재한다면 나중에 코드 리팩토링과 같은 수정이 일어났을 때, 그 기능만을 테스트해볼 수 있어서 아주 좋으니 도전해보시는 것을 추천드려요!!

Trudy2645 committed:
feat: 회원 기능 및 '좋아요', 프로필 API 구현

               SNS의 핵심 기능인 회원 시스템을 추가합니다.
               이제 모든 게시글과 댓글에 작성자 정보가 포함되며, '좋아요'와 회원 프로필 조회가 가능합니다.

               Member (회원가입, 닉네임 검색) 기능 추가

               PostLike ('좋아요' 토글) 기능 추가

               회원 프로필 (작성 글/댓글/좋아요 목록) 조회 API 추가

               기존 Post, Comment에 작성자 연동 및 권한 확인 로직 추가
